R/RunTcrdist3.R

```diff
@@ -70,43 +70,68 @@ RunTcrdist3 <- function(seuratObj,
                       "', debug ='", ifelse(!is.null(debugTcrdist3) && debugTcrdist3, yes = 'True', no = 'False'),
                       "')")
     readr::write_file(command, script, append = TRUE)
-    system2(pythonExecutable, script)
+    pythonOutput <- system2(pythonExecutable, script, stdout = TRUE, stderr = TRUE)
+    if (length(pythonOutput) > 0) {
+      message(paste(pythonOutput, collapse = "\n"))
+    }
+    
+    #ensure cell_df vs clone_df order validation occured in the python layer
+    chain_tcrdist3 <- .convert_chain_for_python(chain)
+    chain_cdr3_id <- tolower(strsplit(chain_tcrdist3, split = "")[[1]][1])
+    expectedValidation <- paste0("validation: row order for cdr3_", chain_cdr3_id, "_aa matches input")
+    if (!any(grepl(expectedValidation, pythonOutput, fixed = TRUE))) {
+      stop(paste0("Python validation message not found in output: ", expectedValidation))
+    }
 
     unlink(input_data_file)
     unlink(script)
 
     #read the RDS files
-    chain_tcrdist3 <- .convert_chain_for_python(chain)
+    
+    #read clone_ids from Python (dimnames set in R since rpy2 dimnames assignment fails)
+    cloneIdsFile <- paste0(rdsOutputPath, "/clone_ids.rds")
+    if (!file.exists(cloneIdsFile)) {
+      stop(paste0("Clone IDs RDS file not found: ", cloneIdsFile))
+    }
+    python_clone_ids <- readRDS(cloneIdsFile)
```

Comment:
I also grab the clone names from python (now that they're validated) and label the distance matrices here. + some extra checks to make sure dat$CloneId and the python clone IDs agree with one another.